From 451856b586d28a576a34fd43215eb10f7f4d52ac Mon Sep 17 00:00:00 2001
From: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Date: Thu, 18 Mar 2021 10:23:05 -0700
Subject: [PATCH 1326/1921] Revert "crypto: marvell: re-enable MBOX interrupt
 in FLR handler"

This reverts commit 0105ca343fd2525da43a14b132cf60ad0617328e.

PF_RST on a FLR is causing RVU_PF_VF_BAR4_ADDR also getting cleared. Due to that RVU_PF_VF_BAR4_ADDR is also getting cleared which would break PF::VF communication

Change-Id: Ic089123ae5402881228d12c3cb70a122299e2b0a
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/48054
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/crypto/marvell/cn10k/cn10k_cptpf_main.c    | 4 ----
 drivers/crypto/marvell/octeontx2/otx2_cptpf_main.c | 4 ----
 2 files changed, 8 deletions(-)

diff --git a/drivers/crypto/marvell/cn10k/cn10k_cptpf_main.c b/drivers/crypto/marvell/cn10k/cn10k_cptpf_main.c
index 858864b5b22c..9e31fb8b38d9 100644
--- a/drivers/crypto/marvell/cn10k/cn10k_cptpf_main.c
+++ b/drivers/crypto/marvell/cn10k/cn10k_cptpf_main.c
@@ -135,10 +135,6 @@ static void cptpf_flr_wq_handler(struct work_struct *work)
 			  RVU_PF_VFTRPENDX(reg), BIT_ULL(vf));
 	cn10k_cpt_write64(pf->reg_base, BLKADDR_RVUM, 0,
 			  RVU_PF_VFFLR_INT_ENA_W1SX(reg), BIT_ULL(vf));
-
-	/* Re-enable MBOX interrupt as it gets cleared in HWVF_RST reset */
-	cn10k_cpt_write64(pf->reg_base, BLKADDR_RVUM, 0,
-			  RVU_PF_VFPF_MBOX_INT_ENA_W1SX(reg), BIT_ULL(vf));
 }
 
 static irqreturn_t cptpf_vf_flr_intr(int __always_unused irq, void *arg)
diff --git a/drivers/crypto/marvell/octeontx2/otx2_cptpf_main.c b/drivers/crypto/marvell/octeontx2/otx2_cptpf_main.c
index 669844aeda36..6455f6bdc64b 100644
--- a/drivers/crypto/marvell/octeontx2/otx2_cptpf_main.c
+++ b/drivers/crypto/marvell/octeontx2/otx2_cptpf_main.c
@@ -141,10 +141,6 @@ static void cptpf_flr_wq_handler(struct work_struct *work)
 			 RVU_PF_VFTRPENDX(reg), BIT_ULL(vf));
 	otx2_cpt_write64(pf->reg_base, BLKADDR_RVUM, 0,
 			 RVU_PF_VFFLR_INT_ENA_W1SX(reg), BIT_ULL(vf));
-
-	/* Re-enable MBOX interrupt as it gets cleared in HWVF_RST reset */
-	otx2_cpt_write64(pf->reg_base, BLKADDR_RVUM, 0,
-			 RVU_PF_VFPF_MBOX_INT_ENA_W1SX(reg), BIT_ULL(vf));
 }
 
 static irqreturn_t cptpf_vf_flr_intr(int __always_unused irq, void *arg)
-- 
2.31.1

